File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }

    public int RemainingMinutesInOven(int elapsedTime)
    {
        return ExpectedMinutesInOven() - elapsedTime;
    }
    
    public int PreparationTimeInMinutes(int layers)
    {
        int prepTimePerLayer = 2;
        return layers * prepTimePerLayer;
    }
    
    public int ElapsedTimeInMinutes(int layers, int elapsedTime)
    {
        return PreparationTimeInMinutes(layers) + elapsedTime;
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven: int = 40
let preparationMinutesPerLayer: int = 2

let remainingMinutesInOven (actualMinutesInOven: int): int = expectedMinutesInOven - actualMinutesInOven

let preparationTimeInMinutes (numberOfLayers: int): int = numberOfLayers * preparationMinutesPerLayer

let elapsedTimeInMinutes (numberOfLayers: int) (actualMinutesInOven: int): int = preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven